api/tests/core/users/test_api.py

```diff
@@ -1769,3 +1777,246 @@ def test_unsuspended_account(self) -> None:
+        assert user_to_anonymise.isActive == False
+        assert user_to_anonymise.irisFrance == None
+        assert user_to_anonymise.validatedBirthDate.day == 1
+        assert user_to_anonymise.validatedBirthDate.month == 1
+        assert user_to_anonymise.roles == [users_models.UserRole.ANONYMISED]
+        assert user_to_anonymise.login_device_history == []
+        assert user_to_anonymise.trusted_devices == []
+        assert len(user_to_anonymise.action_history) == 1
+        assert user_to_anonymise.action_history[0].actionType == history_models.ActionType.USER_ANONYMISED
+        assert user_to_anonymise.action_history[0].authorUserId == user_to_anonymise.id
+        assert user_to_anonymise.action_history[0].actionType == history_models.ActionType.USER_ANONYMISED
+
+    def test_anonymise_non_pro_non_beneficiary_user_iris_not_found(self) -> None:
+        user_to_anonymise = users_factories.UserFactory(
+            firstName="user_to_anonymise",
+            lastConnectionDate=datetime.datetime.utcnow() - datetime.timedelta(days=((3 * 365) + 1)),
+            validatedBirthDate=datetime.datetime.today(),
+        )
+
+        with mock.patch("pcapi.core.users.api.get_iris_from_address", side_effect=geography_exceptions.AddressNotFound):
+            users_api.anonymise_non_pro_non_beneficiary_users(force=False)
+
+        db.session.refresh(user_to_anonymise)
+
+        assert len(batch_testing.requests) == 0
+        assert user_to_anonymise.firstName == "user_to_anonymise"
+
+    def test_anonymise_non_pro_non_beneficiary_user_no_addr_api(self) -> None:
+        user_to_anonymise = users_factories.UserFactory(
+            firstName="user_to_anonymise",
+            lastConnectionDate=datetime.datetime.utcnow() - datetime.timedelta(days=((3 * 365) + 1)),
+            validatedBirthDate=datetime.datetime.today(),
+        )
+
+        with mock.patch("pcapi.core.users.api.get_iris_from_address", side_effect=geography_exceptions.AddressApiError):
+            users_api.anonymise_non_pro_non_beneficiary_users(force=False)
+
+        db.session.refresh(user_to_anonymise)
+
+        assert len(batch_testing.requests) == 0
+        assert user_to_anonymise.firstName == "user_to_anonymise"
+
+    def test_anonymise_non_pro_non_beneficiary_user_keep_email_used_in_offer(self) -> None:
+        user_to_anonymise = users_factories.UserFactory(
+            firstName="user_to_anonymise",
+            lastConnectionDate=datetime.datetime.utcnow() - datetime.timedelta(days=((3 * 365) + 1)),
+            validatedBirthDate=datetime.datetime.today(),
+        )
+        offers_factories.OfferFactory(bookingEmail=user_to_anonymise.email)
+
+        self.import_iris()
+        iris = geography_models.IrisFrance.query.first()
+
+        with mock.patch("pcapi.core.users.api.get_iris_from_address", return_value=iris):
+            users_api.anonymise_non_pro_non_beneficiary_users(force=False)
```

Comment:
[détail] Ici est dans les autres tests plus bas, on pourrait simplifier et éviter ce mock en utilisant `force=True`.

```suggestion
        users_api.anonymise_non_pro_non_beneficiary_users(force=True)
```

Cela rendrait les tests plus simples à lire.

Reply:
Le mock reste necessaire pour ne pas envoyer de requete externe (pour l'instant, a voir ce que ça donne avec `pcapi.connectores.api_adresse`)